### ISSUE TITLE
Batch update router settings on traffic signals network

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#5847]
> What application are you using?

Other / Not Sure

> Describe the problem.

Brian needs assistance running a Python script to batch update network router settings over telnet. He has sent John a list of router IPs to be updated. John has an old version of this script that can be repurposed.

For each router we need to:
- Update login password (John has current and future passwords)
- Enable SSH login
- Enable https web server and disable http web server

We should test this with the router at Cameron Rd and Cross Park Dr. If something goes awry we can physically reset the router.

> How soon do you need this?

Flexible  — An extended timeline is OK

> Is there anything else we should know?

Flexible timeline is ok. 

> Requested By
Brian C.

Request ID: DTS21-101900

**Comment by johnclary** [comment]
@amenity @mateoclarke let's try to tackle this in May 5th sprint. I can walk a dev through it. The code will need to run from one of our on-prem linux servers—those have access to the traffic signal network.

**Comment by johnclary** [comment]
Just reached out to Brian C. to confirm this is still needed.

**Comment by johnclary** [comment]
i would like to get this assigned this sprint

**Comment by sergiogcx** [comment]
The script is mostly finished, I have also extended it to keep track of the devices that are reachable and wether or not those devices can be accessed using username and password.

**Comment by sergiogcx** [comment]
The script has found that we can't ping about 200 devices, still waiting on the telnet access test but so far it looks like we can login to most of them.

![2021-09-15_15-48-55.png](https://images.zenhubusercontent.com/5b7edad7290aac725aec290c/1621bfbc-a7f9-4890-ac41-0548a1ed0cee)

**Comment by amenity** [comment]
@sergiogcx - any updates here?

**Comment by sergiogcx** [comment]
@amenity Yes, let me post an update of the state of things.

I basically had it all wrapped up with about 700 responsive hosts with SSH and HTTPS enabled, minus telnet and HTTP. Also, the password was changed for these according to this doc:
https://docs.google.com/document/d/1LsAFg5JeN-FnxC0utGt7niXDXIQkdXUt8X98Qz_nSVM/edit

Unfortunately, I am told the password in that document is wrong. I already had changed it for 700 devices and I also disabled telnet for these devices so I had to go back and write the same code in SSH and tried resetting the password. Strangely, some of the nodes are not accepting the new password and about 300 have been updated to what it should be, another 400 need to be updated but somehow the SSH script does not work.

If necessary, I am just going to do this manually if it's the last thing I ever do.
